external/fv3fit/fv3fit/reservoir/model.py

```diff
@@ -49,39 +43,19 @@ def dump(self, path: str) -> None:
         Args:
             path: a URL pointing to a directory
         """
-
-        fs: fsspec.AbstractFileSystem = fsspec.get_fs_token_paths(path)[0]
-        fs.makedirs(path, exist_ok=True)
-        mapper = fs.get_mapper(path)
-
-        mapper[self._READOUT_NAME] = self.readout.dumps()
-        metadata = {
-            "reservoir_hyperparameters": dataclasses.asdict(
-                self.reservoir.hyperparameters
-            )
-        }
-        mapper[self._METADATA_NAME] = yaml.safe_dump(metadata).encode("UTF-8")
+        self.readout.dump(path)
+        self.reservoir.dump(f"{path}/{self._RESERVOIR_SUBDIR}")
 
     @classmethod
     def load(cls, path: str) -> "ReservoirComputingModel":
         """Load a model from a remote path"""
-        mapper = fsspec.get_mapper(path)
-
-        f = io.BytesIO(mapper[cls._READOUT_NAME])
-        readout_components = joblib.load(f)
-        readout = ReservoirComputingReadout(**readout_components)
-        metadata = yaml.safe_load(mapper[cls._METADATA_NAME])
-
-        reservoir_hyperparameters = dacite.from_dict(
-            ReservoirHyperparameters, metadata["reservoir_hyperparameters"]
-        )
-
-        return cls(reservoir=Reservoir(reservoir_hyperparameters), readout=readout,)
+        readout = ReservoirComputingReadout.load(path)
+        reservoir = Reservoir.load(f"{path}/{cls._RESERVOIR_SUBDIR}")
```

Comment:
nit (non-blocking): leave this for another PR, but it feels odd that one of these is working in a subdirectory and the other is not.